qiskit/algorithms/minimum_eigensolvers/adapt_vqe.py

```diff
@@ -0,0 +1,311 @@
+    ansatz from a set of evolution operators. It iteratively extends the ansatz circuit, by
+    selecting the building block that leads to the largest gradient from a set of candidates. In
+    chemistry, this is usually a list of orbital excitations. This results in a wavefunction ansatz
+    which is uniquely adapted to the operator whose minimum eigenvalue is being determined.
+    This class relies on a supplied instance of :class:`~.VQE` to find the minimum eigenvalue.
+    The performance of AdaptVQE significantly depends on the minimization routine.
+
+    .. code-block:: python
+
+      from qiskit.algorithms.minimum_eigensolvers import AdaptVQE, VQE
+      from qiskit.algorithms.optimizers import SLSQP
+      from qiskit.primitives import Estimator
+      from qiskit.circuit.library import EvolvedOperatorAnsatz
+
+      # get your Hamiltonian
+      hamiltonian = ...
+
+      # construct your ansatz
+      ansatz = EvolvedOperatorAnsatz(...)
+
+      vqe = VQE(Estimator(), ansatz, SLSQP())
+
+      adapt_vqe = AdaptVQE(vqe)
+
+      eigenvalue, _ = adapt_vqe.compute_minimum_eigenvalue(hamiltonian)
+
+
+    Attributes:
+        solver: a :class:`~.VQE` instance used internally to compute the minimum eigenvalues.
+            It is a requirement that the :attr:`~.VQE.ansatz` of this solver is of type
+            :class:`qiskit.circuit.library.EvolvedOperatorAnsatz`.
+        threshold: the convergence threshold for the algorithm. Once all gradients have an absolute
+            value smaller than this threshold, the algorithm terminates.
+        max_iterations: the maximum number of iterations for the adaptive loop. If ``None``, the
+            algorithm is not bound in its number of iterations.
+    """
+
+    def __init__(
+        self,
+        solver: VQE,
+        *,
+        threshold: float = 1e-5,
+        max_iterations: int | None = None,
+    ) -> None:
+        """
+        Args:
+            solver: a :class:`~.VQE` instance used internally to compute the minimum eigenvalues.
+                It is a requirement that the :attr:`~.VQE.ansatz` of this solver is of type
+                :class:`qiskit.circuit.library.EvolvedOperatorAnsatz`.
+            threshold: the convergence threshold for the algorithm. Once all gradients have an
+                absolute value smaller than this threshold, the algorithm terminates.
+            max_iterations: the maximum number of iterations for the adaptive loop. If ``None``, the
+                algorithm is not bound in its number of iterations.
+        """
+        validate_min("threshold", threshold, 1e-15)
+
+        self.solver = solver
+        self.threshold = threshold
+        self.max_iterations = max_iterations
+        self._tmp_ansatz = self.solver.ansatz
+        self._excitation_pool: list[OperatorBase] = []
+        self._excitation_list: list[OperatorBase] = []
+
+    @property
+    def initial_point(self) -> Sequence[float] | None:
```

Comment:
These are to allow this to be VariationalAlgo right. I do not know how important that is going forwards. I think its ok - maybe we should note either here, or in the main class docstring that this is getting/updating the underlying solver initial point.